### PR TITLE
Allow deactivation of automatic color styling

### LINKED ISSUE #36: Allow the definition of colors through styling instead of configuration
It would be nice if I could create the component in a way that would make it not set color styles from code, e.g.:

```html
<toggle-button :color="null">
```
```css
.vue-js-switch .v-switch-core {
  background: $color-grey
}
.vue-js-switch.toggled .v-switch-core {
  background: $color-active
}
```

### PR DESCRIPTION
**Pull request by lehni**
Closes #36

### CODE REVIEW

File: src/Button.vue [hunk 1]
```diff
@@ -131,23 +136,23 @@ export default {
...
-        ? color.checked
-        : constants.colorChecked
+      let { color, cssColors } = this
+      return cssColors
+        ? undefined
+        : typeof color !== 'object'
+          ? color || constants.colorChecked
+          : contains(color, 'checked')
+            ? color.checked
+            : constants.colorChecked
```
Comment (euvl) [inline src/Button.vue]: Chaining ternary operators is an anti-pattern in javascript. Never use more than one.

File: src/Button.vue [hunk 2]
```diff
@@ -131,23 +136,23 @@ export default {
...
-
-      return contains(color, 'unchecked')
-        ? color.unchecked
-        : constants.colorUnchecked
+      let { color, cssColors } = this
+      return cssColors
+        ? undefined
+        : contains(color, 'unchecked')
+          ? color.unchecked
+          : constants.colorUnchecked
```
Comment (euvl) [inline src/Button.vue]: Same here.

**Comment by euvl** [comment]
I don't think that it is the right place to do it at all.
These methods start returning values of various types, sometimes undefined, etc.


I would rather put a simple check in here:

https://github.com/euvl/vue-js-toggle-button/blob/master/src/Button.vue#L103

Something like:

```
backgroundColor: this.cssColors ? null :  this.colorCurrent,
```

Pretty much thats the only change that should be done to old code.

File: src/Button.vue (hunk 1 above)
Comment (lehni) [reply src/Button.vue]: Who says so? But sure, I'll change it.

**Comment by lehni** [comment]
You're right. The previous commit was a left over of the approach in the first attempt + programming before coffee this morning. This is much cleaner now. Thanks!

**Comment by euvl** [comment]
Nice, great job 👍

**Comment by lehni** [comment]
Very cool, thanks! And take your time with releasing this.

**Comment by lehni** [comment]
@euvl before you deploy a new version, you need to run `npm run build` and commit the changes because my two PRs were both in relation to the master branch, and thus they exclude each other in their state of the dist folder. A better idea may be to define `npm run build` automatically as the `prepare` script and exclude the comited versions in `dist` through `.gitignore`, see https://docs.npmjs.com/misc/scripts

**Comment by euvl** [comment]
Oh, dont worry, i will rebuild it anyways. I had a case on another project i was working on, where someone tried to commit a change with bitcoin miner in dist file 😄 

Some people use plugin directly from Github repo, for example, in china npm sometimes is not an option (as far as i was told), i don't remember other cases, but having dist in main repo is a good thing for plugin 😄

**Comment by euvl** [comment]
Btw I got my laptop yesterday night so will be able to publish this one tonight.

**Comment by lehni** [comment]
@euvl friendly reminder to not forget to publish this. And I hope the new 💻 is good!

**Comment by lehni** [comment]
@euvl any chance to publish a new version? Sorry for keep bugging you...

**Comment by euvl** [comment]
Wow, i completely forgot about it, will do tonight. THanks 👍  😄

**Comment by lehni** [comment]
👋

**Comment by lehni** [comment]
@euvl another reminder that this still hasn't been taken care of. Could you please release a new version? Thanks.

**Pull request merged**